### ISSUE TITLE
Trying to locate problem with using Wx::Choicebook.get_control_sizer

### ISSUE THREAD

**Issue by Randalphwa** [mcorino/wxRuby3#199]
I'm trying to track down a problem with `Wx::Choicebook` when I use get_control_sizer to add an item. Essentially, my code is this:

```Ruby
    @choicebook = Wx::Choicebook.new(page, Wx::ID_ANY)
    page_sizer.add(@choicebook, Wx::SizerFlags.new.border(Wx::ALL))

    btn = Wx::Button.new(@choicebook, Wx::ID_ANY, 'First')
    @choicebook.get_control_sizer.add(btn,
      Wx::SizerFlags.new.expand.border(Wx::ALL))
```

If I leave out the `btn =` and `get_control_sizer` lines the dialog this is in runs fine, and the main window frame that invoked it also continues to run fine. However, when I include those two lines, the dialog runs fine, but when it closes, ruby crashes with `terminated with exit code: 3221226356` which I gather is a heap corruption.

Am I once again doing something obviously wrong?

Also on a related question, I find that breaking into the pry debugger using binding.pry basically locks up even if I remove those two lines. Do you have suggestions for a debugger that would allow breaking into the debugger in a class initialization inherting from a wxWidgets class? I'm trying to find a better way to find which problems are my own code issues before annoying you with a pointless issue...

**Comment by mcorino** [comment]
Good for you! You already discovered one of the leftover holes I mentioned in the release notes :stuck_out_tongue_winking_eye: 
Luckily this one was fairly easy to locate and reproduce. FIx is in.

I'll probably put out a minor update release in 1-2 weeks.

Regarding your debugging question I'm afraid I have to say I have not looked into this at all up to now.
I'm also not sure this is ever going to work with a Ruby debugger. In my experience mixing multiple runtime environments competing for input events is never easy and often (practically) impossible. Do you have experience with something like this working with wxPython?

**Comment by Randalphwa** [comment]
------- Original Message -------
On Thursday, October 19th, 2023 at 5:02 AM, mcorino ***@***.***> wrote:

> Good for you! You already discovered one of the leftover holes I mentioned in the release notes 😜
> Luckily this one was fairly easy to locate and reproduce. FIx is in.
>
> I'll probably put out a minor update release in 1-2 weeks

I have a test app where I try to cover most of the UI that wxUiEditor supports which I use for verifying C++ and wxPython code generation. It has a lot of bundles and svg files, so with wxRuby3 0.8, I wasn't able to verify wxRuby code generation. With the 0.9 release, I have started verifying the wxRuby3 code generation -- and hit this problem in one of the test dialogs that covers all the possible wxWidget books. I just fixed the code to embed SVG files into wxRuby3 code, so I should be able to complete the rest of the verification by early next week. My hope is to release wxUiEditor 1.3 with it's wxRuby3 support in mid November -- so I'll probably set 0.9.1 as the minimum required version.

> Regarding your debugging question I'm afraid I have to say I have not looked into this at all up to now.
>
> I'm also not sure this is ever going to work with a Ruby debugger. In my experience mixing multiple runtime environments competing for input events is never easy and often (practically) impossible. Do you have experience with something like this working with wxPython?

I do have a Python debugger that runs within VS Code that I use as my editor, but I've been fortunate enough that I rarely needed it. There was an VS Code extension providing a debugger for Ruby, but it's been abandoned and apparently no longer works. I tried the pry debugger just to try to find what was causing the crash (initially it was wxUE code generation). At first it worked fine, by letting the program exit and having Ruby display the syntax error. However, once I fixed my code generation, the debugger stopped working.

I only program enough in Python and Ruby to be able to verify wxUE code generation, so the need for a debugger has so far been minimal to none for both languages. In most cases, the problems result from getting Python and Ruby naming conventions mixed up in the code generation and that's easy to spot if I carefully read the Ruby syntax error information before creating an issue in your repo. I do appreciate you gently suggesting that I RTFM or in this case the paragraph of Ruby output that contains buried within it the actual problem. 🫢
